cfgmgr/vxlanmgr.cpp

```diff
@@ -409,9 +470,138 @@ bool VxlanMgr::doVxlanTunnelMapCreateTask(const KeyOpFieldsValuesTuple & t)
+    if (m_vlanMapCache.find(vlan) != m_vlanMapCache.end())
+    {
+        SWSS_LOG_ERROR("Vlan %s already mapped. Map Create failed for : %s", 
+                      vlan.c_str(), vxlanTunnelMapName.c_str());
+        return true;
+    }
+
+    if (m_vniMapCache.find(vni_id) != m_vniMapCache.end())
+    {
+        SWSS_LOG_ERROR("VNI %s already mapped. Map Create failed for : %s", 
+                      vni_id.c_str(), vxlanTunnelMapName.c_str());
+        return true;
+    }
+
+    const auto vlan_prefix = std::string("Vlan");
+    const auto prefix_len = vlan_prefix.length();
+    vlan_id = vlan.substr(prefix_len);
+
+    size_t found = vxlanTunnelMapName.find(delimiter);
+    const auto vxlanTunnelName = vxlanTunnelMapName.substr(0, found);
+
+    // If the vxlan tunnel has been created
+    auto it = m_vxlanTunnelCache.find(vxlanTunnelName);
+    if (!isTunnelActive(vxlanTunnelName))
+    {
+        SWSS_LOG_INFO("Vxlan tunnel %s has not been created", vxlanTunnelName.c_str());
+        // Suspend this message until the vxlan tunnel is created
+        return false;
+    }
+
+    if (!isVlanStateOk(vlan))
+    {
+        SWSS_LOG_INFO("VLAN id is not yet created : %s",vxlanTunnelMapName.c_str());
+        return false;
+    }
+
+    // Check the below condition only after the vxlanmgrd has reached reconcile state
+    // The check to verify the state vxlan table is to take care of back to back 
+    // create and delete of a VTEP object. On deletion of a VTEP object the FRR takes 
+    // some time to remove all the routes and once all the routes are removed, the p2p 
+    // tunnel is also removed. This check waits for all the p2p tunnels which were associated
+    // with the earlier version of the VTEP to be deleted before processing further map entry 
+    // creations.
+    WarmStart::WarmStartState state;
+    WarmStart::getWarmStartState("vxlanmgrd",state);
+    if (state == WarmStart::RECONCILED)
+    {
+        if (m_vxlanTunnelMapCache.empty())
+        {
+            std::vector<std::string> keys;
+            m_stateVxlanTunnelTable.getKeys(keys);
+            if (!keys.empty())
+            { 
+                SWSS_LOG_WARN("State VXLAN tunnel table not yet empty.");
+                return false;
+            }
+        }
+    }
+
+    auto sourceIp = std::find_if(
+        it->second.fvt.begin(),
+        it->second.fvt.end(),
+        [](const FieldValueTuple & fvt){ return fvt.first == SOURCE_IP; });
+    if (sourceIp  == it->second.fvt.end())
+    {
+        SWSS_LOG_DEBUG("Vxlan tunnel %s has no field src_ip", vxlanTunnelName.c_str());
+        return true;
+    }
+    else
+    {
+        src_ip = sourceIp->second;
+    }
+    auto dstIp = std::find_if(
+        it->second.fvt.begin(),
+        it->second.fvt.end(),
+        [](const FieldValueTuple & fvt){ return fvt.first == DST_IP; });
+    if (dstIp  != it->second.fvt.end())
+    {
+        dst_ip = dstIp->second;
+    }
+    else
+    {
```

Comment:
The dst_ip is that of the VXLAN_TUNNEL object. For EVPN it is expected to be NULL. Since the non EVPN cases had the support for destination IP in the VXLAN_TUNNEL object, this check was added.